File: apollo-api/src/main/java/com/apollographql/apollo/api/cache/http/HttpCacheRecordEditor.java
package com.apollographql.apollo.api.cache.http;

import java.io.IOException;

import javax.annotation.Nonnull;

import okio.Sink;

public interface HttpCacheRecordEditor {
  @Nonnull Sink headerSink();

  @Nonnull Sink bodySink();

  void abort() throws IOException;

  void commit() throws IOException;
}

File: apollo-api/src/main/java/com/apollographql/apollo/api/cache/http/HttpCacheRecordEditor.kt
package com.apollographql.apollo.api.cache.http;

import java.io.IOException;

import org.jetbrains.annotations.NotNull;

import okio.Sink;

public interface HttpCacheRecordEditor {
  @NotNull Sink headerSink();

  @NotNull Sink bodySink();

  void abort() throws IOException;

  void commit() throws IOException;
}
